capstone/src/components/User/Adventurer/ExperienceInfo/ExperienceInfo.jsx

```diff
@@ -18,23 +18,30 @@ const API_KEY = process.env.REACT_APP_GOOGLE_MAPS_API_KEY;
 const baseURL = process.env.REACT_APP_BASE_URL;
 
 async function submitRate(username, experienceId, reviews) {
-  const values = { username, experienceId, reviews };
-  return axios.post(`${baseURL}/review/rate`, values);
+  return axios.post(
+    `${baseURL}/adventure/rate`,
+    { reviews },
+    { headers: { username, experienceId } },
+  );
 }
 
 function getPhotoReference(restaurant) {
   return restaurant.photos ? restaurant.photos[0].photo_reference : null;
 }
 
-export default function ExperienceInfo({ onSubmit }) {
+export default function ExperienceInfo({ onUpdateRestaurants }) {
   const { username } = useContext(UserContext);
   const { restaurants } = useContext(AdventurerContext);
+  // All feedback information (including distance, which is not rateable)
   const feedbackInfo = useContext(FeedbackContext);
   const [indexRestaurant, setIndexRestaurant] = useState(0);
   const [newReview, setNewReview] = useState([]);
   // Get the restaurant viewing now
   const currentRestaurant = restaurants.length > 0 ? restaurants[indexRestaurant] : null;
-  const isRated = currentRestaurant?.review != null;
+  // QUESTION: Why is it not responding to changes (derived state)
+  // currentRestaurant.review not updated by setter. Queued. Till when will it be triggered?
+  // Thinking about not a derived state but another independent state for manipulation
```

Comment:
You could also comment on your own PRs with the context instead of adding it as a comment (otherwise you'll have to go back and delete the comment :D)

Reply:
Oh, great! The reason behind adding this QUESTION tags was so that in our 1:1 I could look for that and remember if I have something pending, but I will start doing that.
Thanks!